### PR TITLE
Not getting error responseBody in remrem-publish, if statusCode is not OK

### PR DESCRIPTION
**Pull request by zburswa**
Implemented code changes to handle error ResponseBody in
/generateAndPublish endpoint

### CODE REVIEW

**Comment by zburswa** [comment]
> Looks good, I've tested locally and now I get the expected 400 if I input malformed JSON event to the endpoint. Are there no tests to be updated for these changes?

There is no need to update the test cases for these changes. All the test cases are working fine.

File: publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java [hunk 1]
```diff
@@ -198,21 +198,21 @@ public ResponseEntity generateAndPublish(@ApiParam(value = "message protocol", r
...
+                return new ResponseEntity(parser.parse(e.getResponseBodyAsString()), HttpStatus.UNAUTHORIZED);
+            }else if (e.getStatusCode().equals(HttpStatus.NOT_ACCEPTABLE)) {
+                return new ResponseEntity(parser.parse(e.getResponseBodyAsString()), HttpStatus.NOT_ACCEPTABLE);
+            }else if (e.getStatusCode().equals(HttpStatus.EXPECTATION_FAILED)) {
+                return new ResponseEntity(parser.parse(e.getResponseBodyAsString()), HttpStatus.EXPECTATION_FAILED);
             }
             else {
-                return new ResponseEntity(parser.parse(RemremPublishServiceConstants.GENERATE_INTERNAL_ERROR), HttpStatus.INTERNAL_SERVER_ERROR);
+                return new ResponseEntity(parser.parse(e.getResponseBodyAsString()), HttpStatus.INTERNAL_SERVER_ERROR);
             }
```
Comment (magnusbaeck) [inline publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java]: While we're at it we could consider getting rid of the repetition altogether, e.g. with something like this:

```Java
HttpStatus result = HttpStatus.INTERNAL_SERVER_ERROR;
EnumSet<HttpStatus> passthrough = EnumSet.of(HttpStatus.BAD_REQUEST, HttpStatus.SERVICE_UNAVAILABLE, ...);
if (passthrough.contains(e.getStatusCode()) {
    result = e.getStatusCode();
}
return new ResponseEntity(parser.parse(e.getResponseBodyAsString()), result);
```

**Comment by e-pettersson-ericsson** [comment]
> > Looks good, I've tested locally and now I get the expected 400 if I input malformed JSON event to the endpoint. Are there no tests to be updated for these changes?
> 
> There is no need to update the test cases for these changes. All the test cases are working fine.

Hmm, yes but that implies we don't have any real tests on this particular endpoint. From what I can see, we only have mocked tests for this endpoint, hence why we didn't discover this issue previously. I think it would be good to add a scenario in integration test for this endpoint, with both good and bad responses.

**Comment by zburswa** [comment]
> > > Looks good, I've tested locally and now I get the expected 400 if I input malformed JSON event to the endpoint. Are there no tests to be updated for these changes?
> > 
> > 
> > There is no need to update the test cases for these changes. All the test cases are working fine.
> 
> Hmm, yes but that implies we don't have any real tests on this particular endpoint. From what I can see, we only have mocked tests for this endpoint, hence why we didn't discover this issue previously. I think it would be good to add a scenario in integration test for this endpoint, with both good and bad responses.

Ok. Will check whether we can add the test scenarios for this endpoint.

File: publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java [hunk 2]
```diff
@@ -198,22 +199,13 @@ public ResponseEntity generateAndPublish(@ApiParam(value = "message protocol", r
...
-            }else if (e.getMessage().startsWith(Integer.toString(HttpStatus.NOT_ACCEPTABLE.value()))) {
-                return new ResponseEntity(parser.parse(RemremPublishServiceConstants.ERLOOKUP_NONEFOUND), HttpStatus.NOT_ACCEPTABLE);
-            }else if (e.getMessage().startsWith(Integer.toString(HttpStatus.EXPECTATION_FAILED.value()))) {
-                return new ResponseEntity(parser.parse(RemremPublishServiceConstants.ERLOOKUP_MULTIPLEFOUND), HttpStatus.EXPECTATION_FAILED);
-            }
-            else {
-                return new ResponseEntity(parser.parse(RemremPublishServiceConstants.GENERATE_INTERNAL_ERROR), HttpStatus.INTERNAL_SERVER_ERROR);
+        catch (HttpStatusCodeException e) {
+            HttpStatus result = HttpStatus.BAD_REQUEST;
+            EnumSet<HttpStatus> getStatus = EnumSet.of(HttpStatus.SERVICE_UNAVAILABLE, HttpStatus.UNAUTHORIZED, HttpStatus.NOT_ACCEPTABLE, HttpStatus.EXPECTATION_FAILED, HttpStatus.INTERNAL_SERVER_ERROR);
```
Comment (erjamni-raja) [inline publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java]: Please initialize _getStatus_ the class level. 
If it is in catch each request to create _getStatus_.

**Comment by zburswa** [comment]
> > > > Looks good, I've tested locally and now I get the expected 400 if I input malformed JSON event to the endpoint. Are there no tests to be updated for these changes?
> > > 
> > > 
> > > There is no need to update the test cases for these changes. All the test cases are working fine.
> > 
> > 
> > Hmm, yes but that implies we don't have any real tests on this particular endpoint. From what I can see, we only have mocked tests for this endpoint, hence why we didn't discover this issue previously. I think it would be good to add a scenario in integration test for this endpoint, with both good and bad responses.
> 
> Ok. Will check whether we can add the test scenarios for this endpoint.

We have the test class"EiffelRemremCommonControllerUnitTest.java" . In this test class both positive and negative scenarios for /generateAndPublish Endpoint are tested.

**Pull request merged**